dao/mongodb.rya/src/test/java/org/apache/rya/mongodb/MongoDBRyaDAOTest.java

```diff
@@ -28,102 +30,279 @@
+
+        // Doc requires "(A|B)&(C|(D&E))" and user has "B" = User CANNOT view
+        assertFalse(testVisibilityStatement("(A|B)&(C|(D&E))", new Authorizations("B")));
+
+        // Doc requires "A|(B&C&(D|E))" and user has "A" = User can view
+        assertTrue(testVisibilityStatement("A|(B&C&(D|E))", new Authorizations("A")));
+
+        // Doc requires "A|(B&C&(D|E))" and user has "B" and "C" and "D" = User can view
+        assertTrue(testVisibilityStatement("A|(B&C&(D|E))", new Authorizations("B", "C", "D")));
+
+        // Doc requires "A|(B&C&(D|E))" and user has "B" and "C" and "E" = User can view
+        assertTrue(testVisibilityStatement("A|(B&C&(D|E))", new Authorizations("B", "C", "E")));
+
+        // Doc requires "A|(B&C&(D|E))" and user has "B" = User CANNOT view
+        assertFalse(testVisibilityStatement("A|(B&C&(D|E))", new Authorizations("B")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "A" = User can view
+        assertTrue(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("A")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "E" = User CANNOT view
+        assertFalse(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("E")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "E" and "F" = User CANNOT view
+        assertFalse(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("E", "F")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "I" = User CANNOT view
+        assertFalse(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("I")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "A" and "I" = User can view
+        assertTrue(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("A", "I")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "E" and "F" and "G" and "H" = User can view
+        assertTrue(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("E", "F", "G", "H")));
+
+        // Doc requires "A|B|C|D|(E&F&G&H)" and user has "E" and "F" and "G" and "H" and "I" = User can view
+        assertTrue(testVisibilityStatement("A|B|C|D|(E&F&G&H)", new Authorizations("E", "F", "G", "H", "I")));
+    }
+
+    /**
+     * Generates a test statement with the provided document visibility to
+     * determine if the specified user authorization can view the statement.
+     * @param documentVisibility the document visibility boolean expression
+     * string.
+     * @param userAuthorizations the user authorization strings.
+     * @return {@code true} if provided authorization could access the document
+     * in the collection. {@code false} otherwise.
+     * @throws RyaDAOException
+     */
+    private boolean testVisibilityStatement(final String documentVisibility, final Authorizations userAuthorizations) throws RyaDAOException {
+        final MongoDatabase db = mongoClient.getDatabase(configuration.get(MongoDBRdfConfiguration.MONGO_DB_NAME));
```

Comment:
hitting mongo makes this an integration test.  can you move these out to another file for visiblity integration tests?